include/seqan3/argument_parser/auxiliary.hpp

```diff
@@ -15,12 +15,160 @@
 #include <sstream>
 #include <vector>
 
+#include <seqan3/core/detail/customisation_point.hpp>
 #include <seqan3/core/detail/debug_stream_type.hpp>
+#include <seqan3/core/type_traits/basic.hpp>
 #include <seqan3/io/stream/concept.hpp>
+#include <seqan3/std/concepts>
+
+namespace seqan3::custom
+{
+
+/*!\brief A type that can be specialised to provide customisation point implementations so that third party types
+ *        model the seqan3::string_convertible concept.
+ * \tparam t The type you wish to specialise for.
+ * \ingroup argument_parser
+ *
+ * \details
+ *
+ * You can specialise a class like this:
+ *
+ * \include test/snippet/argument_parser/string_convertible.cpp
+ *
+ * Please note that by default the `t const`, `t &` and `t const &` specialisations of this class inherit the
+ * specialisation for `t` so you usually only need to provide a specialisation for `t`.
+ *
+ * \note Only use this if you cannot provide respective functions in your namespace.
+ */
+template <typename t>
+struct string_convertibility
+{}; // forward
+
+//!\cond
+template <typename t>
+struct string_convertibility<t const> : string_convertibility<t>
+{};
+
+template <typename t>
+struct string_convertibility<t &> : string_convertibility<t>
+{};
+
+template <typename t>
+struct string_convertibility<t const &> : string_convertibility<t>
+{};
+//!\endcond
+
+} // seqan3::custom
+
+namespace seqan3::detail::adl_only
+{
+
+//!\brief Poison-pill overload to prevent non-ADL forms of unqualified lookup.
+template <typename t>
+std::unordered_map<std::string, t> string_conversion_map(t v) = delete;
```

Comment:
I don't think it's good style to name a function "string_conversion_map"...

Reply:
I told you I need a good name :grin:

Reply:
@h-2 Can you make a naming suggestions?

Reply:
> Maybe call it named_values? Name-value-pair and attribute-value-pair are popular names for this, I think... it also depends on the order of key/value... maybe also value_to_name_map.